From 064bad4ab17a9bf4540073038df291e884b18b7e Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 5 Sep 2014 19:49:06 -0400
Subject: [kernel] crash_dump: Make is_kdump_kernel() accessible from modules

Message-id: <1409946615-5862-2-git-send-email-mschmidt@redhat.com>
Patchwork-id: 91545
O-Subject: [RHEL7.1 PATCH 01/70] crash_dump: Make is_kdump_kernel() accessible from modules
Bugzilla: 1089402 1059554
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1059554
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1089402

Upstream commit:
commit b3292e88e33677a90c18236fd895ace2d80efa3c
Author: Amir Vadai <amirv@mellanox.com>
Date:   Mon Aug 25 16:06:52 2014 +0300

    crash_dump: Make is_kdump_kernel() accessible from modules

    In order to make is_kdump_kernel() accessible from modules, need to
    make elfcorehdr_addr exported.
    This was rejected in the past [1] because reset_devices was prefered in
    that context (reseting the device in kdump kernel), but now there are
    some network drivers that need to reduce memory usage when loaded from
    a kdump kernel.  And in that context, is_kdump_kernel() suits better.

    [1] - https://lkml.org/lkml/2011/1/27/341

    CC: Vivek Goyal <vgoyal@redhat.com>
    Signed-off-by: Amir Vadai <amirv@mellanox.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/kernel/crash_dump.c b/kernel/crash_dump.c
index c766ee5..b64e238 100644
--- a/kernel/crash_dump.c
+++ b/kernel/crash_dump.c
@@ -18,6 +18,7 @@ unsigned long saved_max_pfn;
  * it under CONFIG_CRASH_DUMP and not CONFIG_PROC_VMCORE.
  */
 unsigned long long elfcorehdr_addr = ELFCORE_ADDR_MAX;
+EXPORT_SYMBOL_GPL(elfcorehdr_addr);
 
 /*
  * stores the size of elf header of crash image
-- 
1.7.1